### ISSUE TITLE
[BUG] Promise.prototype.constructor is read-only with @inject-into content

### ISSUE THREAD

**Issue by bipface** [violentmonkey/violentmonkey#1480]
tested with violentmonkey v2.13.0.17 with firefox v91.4.1ESR and v99.0b8.
violentmonkey v2.13.0.16 doesn't appear to be affected.

**To Reproduce**
- install the script below
- visit <https://violentmonkey.github.io/>
- expect: ` success` is logged
- actual: `TypeError: "constructor" is read-only`

scripts using `@inject-into page` aren't affected.

```js
// ==UserScript==
// @name        test-promise
// @namespace   Violentmonkey Scripts
// @match       https://violentmonkey.github.io/*
// @grant       none
// @version     1.0
// @inject-into content
// ==/UserScript==

'use strict';

let p = new Promise(() => {});
p.constructor = {};

console.log('success');

```

alternatively, this script:

```js
// ==UserScript==
// @name        test-require-corejs
// @namespace   Violentmonkey Scripts
// @match       https://violentmonkey.github.io/*
// @require     https://cdn.jsdelivr.net/npm/core-js-bundle@3.21.1/index.js
// @grant       none
// @version     1.0
// @inject-into content
// ==/UserScript==

console.log('success');
```

some scripts (not written by me) rely on this 'core-js' library which seems to attempt to assign to `.constructor` during its initialisation.

I couldn't find any other classes with this quirk, using the following test:

```js
for (let k of [
	'Object',
	'Boolean',
	'Symbol',
	'Error',
	'Number',
	'BigInt',
	'Date', 
	'String',
	'RegExp',
	'Array',
	'Uint8Array',
	'Map',
	'Set',
	'WeakMap',
	'WeakSet',
	'Promise',])
{
	let proto = globalThis[k].prototype;
	
	console.log(k, 'ctor is writable:',
		Reflect.getOwnPropertyDescriptor(proto, 'constructor').writable);
}
```

**Comment by tophf** [comment]
Fixed in 85ce0176.

**Comment by bipface** [comment]
thanks -- can confirm it's been fixed
